grpc/cargo_casper/tests/integration_tests.rs

```diff
@@ -71,6 +71,9 @@ fn run_tool_and_resulting_tests(turbo: bool) {
     test_cmd.arg("test").current_dir(test_dir.join("tests"));
     let test_output = output_from_command(test_cmd);
     assert_eq!(SUCCESS_EXIT_CODE, test_output.status.code().unwrap());
```

Comment:
If the temporary directory should be persisted, you'd check for `test_output.status.code()== SUCCESS_EXIT_CODE` here and if not equal, call `test_dir.into_path()` and output it with a prominent message.

Reply:
Also, the more conventional way of checking for this is to call `status.success()` instead of manually comparing the code.

Reply:
Agreed when only test code can panic, but `output_from_command` can panic too if the tmp build fails. It's bigger change now though - https://github.com/CasperLabs/casperlabs-node/blob/ae575e1f8712e1c6cc5a5af746afa437d3776af0/grpc%2Fcargo_casper%2Ftests%2Fintegration_tests.rs#L39-L45

Reply:
I wouldn't worry about the `status.success()` part just to be clear (I don't really care either way - what's currently there has the benefit of symmetry with the error check earlier, but Marc's also correct).

So, we're only talking about the issue of retaining the temp dir if the test fails.

If we want to do that in a robust, future-proof way, I believe we should use a RAII type for the dir which cleans up on drop unless the thread is panicking.

I'd also be fine with just always cleaning up the temp dir, since I'm not sure this is worth the effort.

Reply:
Never mind - I missed the `into_path()` at the top of the test.  Sorry!